external/avro/src/main/scala/org/apache/spark/sql/avro/AvroSerializer.scala

```diff
@@ -201,13 +201,11 @@ class AvroSerializer(rootCatalystType: DataType, rootAvroType: Schema, nullable:
 
   private def newStructConverter(
       catalystStruct: StructType, avroStruct: Schema): InternalRow => Record = {
-    if (avroStruct.getType != RECORD) {
+    if (avroStruct.getType != RECORD || avroStruct.getFields.size() != catalystStruct.length) {
       throw new IncompatibleSchemaException(s"Cannot convert Catalyst type $catalystStruct to " +
         s"Avro type $avroStruct.")
     }
-    val avroFields = avroStruct.getFields
-    assert(avroFields.size() == catalystStruct.length)
```

Comment:
Is this check redundant? Why remove it now?

Reply:
I think he moved this condition above.

Reply:
Ah, yea, strictly sounds unrelated tho.

Reply:
Ah, I miss it. :)